iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/semaphore_interface.hpp

```diff
@@ -43,10 +43,6 @@ class SemaphoreInterface
     ///         semaphore was removed from outside the process
     cxx::expected<SemaphoreError> post() noexcept;
 
-    /// @brief Returns the value of the semaphore
-    /// @return Fails when semaphore was removed from outside the process
-    cxx::expected<uint32_t, SemaphoreError> getValue() noexcept;
```

Comment:
Since `sem_getvalue` causes a lot of problems and I thought that our semaphore wrappers use the smallest common subset of posix functionality. It is impossible to implement `sem_getvalue` for named semaphores on mac with our current approach.
When we would now use this feature during our development we could quickly end up with code which does not work on mac so for me the safe approach was to not support this feature at all.

Currently we are also not really using it.